tests/integration/test_parallel_inception.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import print_function, unicode_literals
+
+import textwrap
+
+from tox.util.main import MAIN_FILE
+
+
+def test_parallel_inception(tmpdir, cmd):
+    # the outer config just has one env: graham
+    content = textwrap.dedent(
+        """
+        [tox]
+        envlist = graham
+        skipsdist = True
+
+        [testenv]
+        commands =
+            python runner.py
+    """
+    )
+    outer_config = tmpdir / "tox.ini"
+    outer_config.write_text(content, encoding="utf-8")
+
+    # the inner config has 3 different envs, 1 of them is graham
+    content = textwrap.dedent(
+        """
+        [tox]
+        envlist = graham,john,terry
+        skipsdist = True
+
+        [testenv]
+        commands =
+            python -c 'pass'
+    """
+    )
+    inner_dir = tmpdir / "inner"
+    inner_dir.mkdir()
+    inner_config = inner_dir / "tox.ini"
+    inner_config.write_text(content, encoding="utf-8")
+
+    # the outer test runs the inner tox and asserts all 3 envs were run
+    content = textwrap.dedent(
+        """
+        import os
+        import subprocess
+        import sys
+
+        os.chdir("inner")
+        p = subprocess.Popen(("tox"), stdout=subprocess.PIPE, universal_newlines=True)
+        stdout, _ = p.communicate()
+        sys.stdout.write(stdout)
+        assert "graham" in stdout
+        assert "john" in stdout
+        assert "terry" in stdout
+    """
+    )
+    runner = tmpdir / "runner.py"
+    runner.write_text(content, encoding="utf-8")
+
+    with tmpdir.as_cwd():
```

Comment:
you can pass this as cwd arg to cmd, no need to change

Reply:
TypeError: run() got an unexpected keyword argument 'cwd'